.github/workflows/publish.yml

```diff
@@ -30,16 +31,31 @@ jobs:
           java-version: '11'
       - name: Set up Gradle
         uses: gradle/gradle-build-action@v3
+
       - name: Configure the project version
-        env:
-          IS_PR: ${{ github.event_name == 'pull_request' }}
+        id: version
         run: |-
-          if [[ "$IS_PR" = 'true' ]]; then
-            echo "0.0.1-SNAPSHOT" > version.txt
+          if [[ "${{ github.event_name }}" == 'pull_request' || "${{ github.event_name }}" == 'workflow_dispatch' ]]; then
+            version="0.0.1-${GITHUB_SHA}-SNAPSHOT"
           else
-            echo "${GITHUB_REF#refs/*/}" > version.txt
+            version="${GITHUB_REF#refs/*/}"
           fi
+
+          echo "${version}" > version.txt
           cat version.txt
+
+          echo "VERSION=${version}" >> $GITHUB_OUTPUT
+      - name: Determine Docker metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            modelix/model-server
+            modelix/modelix-model
+          tags: |
+            type=raw,value=${{ steps.version.outputs.VERSION }},enable=true
```

Comment:
> /usr/bin/docker buildx build --file ./model-server/Dockerfile --iidfile /home/runner/work/_temp/docker-actions-toolkit-0IDqvj/iidfile --label org.opencontainers.image.created=2024-02-12T13:13:22.169Z --label org.opencontainers.image.description=Mostly model related components --label org.opencontainers.image.licenses=Apache-2.0 --label org.opencontainers.image.revision=869a63a6c43458e09a9e7edefe30ddfce6ac3ef7 --label org.opencontainers.image.source=https://github.com/modelix/modelix.core --label org.opencontainers.image.title=modelix.core --label org.opencontainers.image.url=https://github.com/modelix/modelix.core --label org.opencontainers.image.version=0.0.1-869a63a6c43458e09a9e7edefe30ddfce6ac3ef7-SNAPSHOT --platform linux/amd64,linux/arm64 --provenance mode=max,builder-id=https://github.com/modelix/modelix.core/actions/runs/7872295160 --tag modelix/model-server:0.0.1-869a63a6c43458e09a9e7edefe30ddfce6ac3ef7-SNAPSHOT --tag modelix/modelix-model:0.0.1-869a63a6c43458e09a9e7edefe30ddfce6ac3ef7-SNAPSHOT --metadata-file /home/runner/work/_temp/docker-actions-toolkit-0IDqvj/metadata-file ./model-server

> modelix/modelix-model:0.0.1-869a63a6c43458e09a9e7edefe30ddfce6ac3ef7-SNAPSHOT

Should we use a shorter commit hash?

On a release tag we should also publish the tag `latest` like in the script.

Reply:
I can shorten that

Reply:
Now we would get `modelix/model-server:0.0.1-bf016f5-SNAPSHOT`

Reply:
That looks better.

What about the `latest` tag for releases?

Reply:
Yes, missed that.

Reply:
updated to include that, too

Reply:
Now the SNAPSHOT builds are also published as `latest`. 
I think we need to distinguish proper release and SNAPSHOT releases.

Reply:
I think we should have another look on this once the next release is due. Might be that something's still of by then.